network/src/connectivity_manager/mod.rs

```diff
@@ -338,6 +362,30 @@ where
             })
             .collect();
 
+        // Limit the number of dialed connections from a Full Node
+        // This does not limit the number of incoming connections
+        // It enforces that a full node cannot have more outgoing connections than `connection_limit`
+        // including in flight dials.
+        let to_connect_size = if let Some(conn_limit) = self.connection_limit {
+            min(
+                conn_limit - self.connected.len() - self.dial_queue.len(),
+                to_connect.len(),
+            )
+        } else {
+            to_connect.len()
+        };
+
+        // If there's nothing to dial, let's just skip the rest of the function
+        if to_connect_size == 0 {
+            return;
+        }
+        to_connect.choose_multiple(&mut self.rng, to_connect_size);
```

Comment:
shouldn't we move this line to the line below in 406

`  for (p, addrs) in to_connect(choose_multiple(&mut self.rng, to_connect_size).iter()`

I think right now, you're effectively randomizing and then throwing away, since the choose_multiple doesn't mutate to_connect.

Reply:
oh, whoops, I thought it said `let to_connect = to_connect.choose_multiple`